sklearn/ensemble/gradient_boosting.py

```diff
@@ -1034,6 +1034,7 @@ def fit(self, X, y, sample_weight=None, monitor=None):
                                  % (self.n_estimators,
                                     self.estimators_.shape[0]))
             begin_at_stage = self.estimators_.shape[0]
+            X = check_array(X, dtype=DTYPE, order="C", accept_sparse='csr')
```

Comment:
Add a comment to say that the requirements of decision_function are more constrained than for fit in general. Disable the finiteness check.

Or just avoid check_array and more explicitly just use tocsr here.